src/electionguard/guardian.py

```diff
@@ -163,18 +169,82 @@ def __init__(
         :param sequence_order: a unique number in [1, 256) that identifies this guardian
         :param number_of_guardians: the total number of guardians that will participate in the election
         :param quorum: the count of guardians necessary to decrypt
-        :param nonce_seed: an optional `ElementModQ` value that can be used to generate the `ElectionKeyPair`.
-                           It is recommended to only use this field for testing.
+        :param election_keys the private keys the guardian generated during a key ceremony
+        :param election_public_keys the public keys the guardian generated during a key ceremony
+        :param partial_key_backups the partial key backups the guardian generated during a key ceremony
         """
         self.id = id
         self.sequence_order = sequence_order
         self.set_ceremony_details(number_of_guardians, quorum)
-        self._backups_to_share = {}
-        self._guardian_election_public_keys = {}
-        self._guardian_election_partial_key_backups = {}
-        self._guardian_election_partial_key_verifications = {}
+        self._backups_to_share = {} if backups_to_share is None else backups_to_share
+        self._guardian_election_public_keys = (
+            {} if election_public_keys is None else election_public_keys
+        )
+        self._guardian_election_partial_key_backups = (
+            {} if partial_key_backups is None else partial_key_backups
+        )
+        self._guardian_election_partial_key_verifications = (
+            {}
+            if guardian_election_partial_key_verifications is None
+            else guardian_election_partial_key_verifications
+        )
+
+        if not election_keys is None:
+            self._election_keys = election_keys
+
+    @classmethod
+    def from_context_info(
```

Comment:
This could be rolled into the `from_nonce`. No reason for an almost duplicate constructor.

Reply:
To expand on this, there are 3 constructors already.

1.  `__init__` , which uses a provided `ElectionKeyPair`
2. `from_nonce`, which generates an ElectionKeyPair using a nonce (or random nonce if none provided)
3. `from_record`, which is post decryption

Reply:
The problem is that this constructor is the one used by the majority of instantiations of Guardian.  It can't be rolled into `from_nonce` because callees don't have a nonce, and it can't be rolled into `from_private_record` because callees don't have a `PrivateGuardianRecord`.  

It could go back into `__init__` if you like, but then we'd have to move the `generate_election_key_pair` call back into there, and put back the if statement so that the other constructors don't call it.  Should I do that?

Reply:
In these cases, they ARE using a nonce. This is a misnomer hidden by the code. They will randomly generate a nonce when none is available. 

Either the key_pair is provided OR a nonce will be used to create one.

https://github.com/microsoft/electionguard-python/blob/03698c683ab35f608e35dc9b66217caa1d40f7eb/src/electionguard/election_polynomial.py#L78

Reply:
merged